src/Parallel/Algorithms/AlgorithmNodegroup.hpp

```diff
@@ -0,0 +1,59 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// This file should be included in each file which defines a nodegroup
+/// parallel component; doing so ensures that the correct charm++ chares are
+/// defined for executables that use that parallel component.
+
+#pragma once
+
+#include "Parallel/Algorithm.hpp"
+#include "Parallel/Algorithms/AlgorithmNodegroupDeclarations.hpp"
+#include "Parallel/ArrayIndex.hpp"
+
+/*!
+ * \ingroup ParallelGroup
+ * \brief A charm++ algorithm singleton chare.
```

Comment:
Since the documentation for all of these is exactly the same except for `singleton`, etc. Is there a way to use something like Doxygen's `copydoc` feature?

Reply:
`copydoc` doesn't look flexible enough to include the small differences. I recall we enabled macro expansion for the dox though, so I could put the doc string and optionally the entire class definition in a macro. How do you feel about that?

Reply:
nevermind, what I suggested no good either. Doxygen appears to require the macro you're expanding to be in the same file, and it does something catastrophic to multiline comments in macros...

Reply:
Hmm, okay. Well no worries! I just thought I'd ask if there is some way of simplifying things :)